app/controllers/course/leaderboards_controller.rb

```diff
@@ -34,6 +44,11 @@ def load_course_users
                     includes(:user)
   end
 
+  # Load approved students from current course with course statistics.
+  def load_course_groups
+    @course_groups = @course.groups
```

Comment:
This should really be part of your action. Strictly speaking the `@groups` instance variable should be set, following the name of the action.